### ISSUE TITLE
Improve cubic bezier curves approximation with quadratic beziers

### ISSUE THREAD

**Issue by puzrin** [fontello/svg2ttf#25]
Current cubic2quad convert algorythm is very basic. That does not affect original fonts, but can cause issues, when originals are drawn in SVG. We use stupid midpoint approach with 2 quad curves. It sucks, because we need to keep tangents on edge points.

1. Approaches:
  - http://www.timotheegroleau.com/Flash/articles/cubic_bezier_in_flash.htm
  - http://stackoverflow.com/a/14514491/1031804
2. Error metrict calculation:
  - ? Hausdorff: http://mathnet.kaist.ac.kr/mathnet/thesis_file/jksiam-v13n4p257.pdf
  - ? Midpoint

Example to reproduce problem (from octicons font):

```svg
<svg height="1024" width="1024" xmlns="http://www.w3.org/2000/svg">
  <path d="M768 768H576c0 0-254.281 0-256-256 0-22.281 3.469-43.469 8.312-64h137.405c-11 18.875-17.719 40.562-17.719 64 0 128 128 128 128 128h192c0 0 128 0 128-128S768 384 768 384s-0.125-64-64-128h64c0 0 256 0 256 256S768 768 768 768zM695.688 576H558.25c11-18.875 17.75-40.562 17.75-64 0-128-128-128-128-128H256c0 0-128 0-128 128s128 128 128 128-3.906 64.875 65.719 128H256c0 0-256 0-256-256s256-256 256-256h192c0 0 256 0 256 256C704 534.281 700.5 555.5 695.688 576z" />
</svg>
```

**Comment by Grawl** [comment]
I have a lot of SVGs (FontAwesome) generated by IcoMoon. I want to pack icon font for my internal company project with some of FontAwesome icons and some own ones. I use `gulp-iconfont` to do it; linked issues about mangled shapes bring me to this issue. What can I do with all my SVGs to get TTF/WOFF/EOT from they? I cannot just redraw them all. I want to pack them without any changing of these beautiful shapes.

**Comment by puzrin** [comment]
> What can I do with all my SVGs to get TTF/WOFF/EOT from they?

You can generate svg only, and convert to ttf via fontforge (it can be scripted too)

**Comment by Grawl** [comment]
Awesome.

**Comment by lukeapage** [comment]
@puzrin do you have any plans to work on this?

**Comment by puzrin** [comment]
Not in nearest future.

**Comment by puzrin** [comment]
Finally fixed in 2.1. Enjoy!

**Comment by Grawl** [comment]
😌
